lang/rust/avro/src/schema.rs

```diff
@@ -271,44 +275,51 @@ impl Name {
             _ => None,
         };
 
-        let aliases: Option<Vec<String>> = complex
-            .get("aliases")
-            .and_then(|aliases| aliases.as_array())
-            .and_then(|aliases| {
-                aliases
-                    .iter()
-                    .map(|alias| alias.as_str())
-                    .map(|alias| alias.map(|a| a.to_string()))
-                    .collect::<Option<_>>()
-            });
-
         Ok(Name {
             name: type_name.unwrap_or(name),
             namespace: namespace_from_name.or_else(|| complex.string("namespace")),
-            aliases,
         })
     }
 
     /// Return the `fullname` of this `Name`
     ///
     /// More information about fullnames can be found in the
     /// [Avro specification](https://avro.apache.org/docs/current/spec.html#names)
-    pub fn fullname(&self, default_namespace: Option<&str>) -> String {
+    pub fn fullname(&self, default_namespace: Namespace) -> String {
         if self.name.contains('.') {
```

Comment:
I believe this is no more possible and the `if` branch can be safely removed.

Reply:
I believe it is still possible in the case of a manually constructed `Name` struct, but I think that probably shouldn't be officially supported flow. If there is ever any appetite to lock down/ put up guardrails to prevent invalid structs from being created, I nominate Name and Schema

Reply:
Good point!
I haven't tried it but it should be possible to prevent manual construction by making the struct and/or its fields (crate) private.
The users should use only pub methods/constructors/getters from the `impl`.

Reply:
Tried to remove and ran into this: https://issues.apache.org/jira/browse/AVRO-3452. felt it deserved its own PR